Add userdef

theAkito opened the pull request:
## What `userdef` does

This tool is a more advanced `adduser` / `useradd` for your [Alpine](https://www.alpinelinux.org/) and [BusyBox](https://www.busybox.net/) based Docker images.

For example, this tool may create a custom OS user with a custom ID inside pre-made Docker images, which perhaps already have a custom user defined and rebuilding the Docker image just to have your custom user in it is not an option.


https://github.com/theAkito/userdef#what

## Why `userdef` exists

### Reason 1
Now, more and more server apps try to go with the current meta of being available on Kubernetes, etc. This is a good idea, however it's often not well executed.
Almost all of the popular server apps are not cloud-native. Their structure is still of some legacy kind.
Examples are [Mattermost](https://mattermost.com/), [Gitea](https://gitea.io/en-us/), [Nextcloud](https://nextcloud.com/).
These server apps have Helm Charts available. However, applying best practices, especially the ones regarding security are not easy to achieve.
Especially, when talking about the `podSecurityContext`. The `fsGroup` option is either not respected properly, which leads to broken deployments, or it's not even available and you have to add it yourself to the Helm Chart.

For example, take the Gitea Helm Chart as an example. You are allowed to set the `podSecurityContext`:
https://gitea.com/gitea/helm-chart/src/commit/d94226765d6e1f197a3112e1b1abbcd73a8bea33/values.yaml#L19-L20

But, if you provide your custom `fsGroup` value, then the deployment will be broken. Why?
https://github.com/go-gitea/gitea/blob/66f2210feca0b50d305a46a203c2b3d2f4d3790b/Dockerfile.rootless#L39-L48

Because the user and group ID of `1000` is hard-coded into the Docker image.

Now, imagine you have an `sshfs` mount, which requires you using the user of the ID `9234`.
The hard-coded `1000` inside the image breaks usage of this `sshfs` mount, just because it does not let you define a custom user with a custom ID.

To make all this work more smoothly, this tool aims to delete the existing user in that Docker image and then recreate it with *your* custom user, which has an ID defined by *you*, instead of being forced to use the randomly chosen hard-coded user ID.

### Reason 2

You may just as well use this tool as a better `adduser` where the actual `adduser` or `useradd` (like the one in [Alpine](https://www.alpinelinux.org/)) have arbitrary and unnecessary restrictions, like for example [limiting the UID/GID size to 256000](https://stackoverflow.com/q/41807026/7061105).


https://github.com/theAkito/userdef#why

agebhar1 reviewed (changes requested):
Hi @theAkito,

thanks for your  contribution - could you please update the file by using a link to your Github account since it's the second tool in the list (see bottom of README.md). Currently the quality check is broken. Thanks :)

theAkito commented:
@agebhar1 

Thanks for the help. 🙂 

Not sure if I understand correctly.

Do you mean something like this:

```
[Oracle][oracle]
```

So, I would need to do this:

```
[Akito][theAkito]
```

?

agebhar1 commented:
@theAkito 
```
[Akito][theAkito]
```
exactly :+1:  So, the URI exists only once. `theAkito` is then on the bottom of the file, like others with more than one occurence.

theAkito commented:
@agebhar1 

I'm still missing something, apparently...

agebhar1 commented:
No worries, add the line to the _bottom_ of the the file (README.md):
```
[Akito]: https://github.com/theAkito
```

theAkito commented:
@agebhar1 

I've added the reference and changed both entries.

Reference
```
[Akito][Akito]
```

Target
```
[Akito]: https://github.com/theAkito
```

agebhar1 commented:
Thanks @theAkito :+1:
